ProcessLib/BoundaryCondition/Python/BHEInflowPythonBoundaryConditionPythonSideInterface.h

```diff
@@ -0,0 +1,105 @@
+/**
+ * \file
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+namespace ProcessLib
+{
+//! Base class for BHENetwork.
+//! This class will get Python bindings and is intended to be to be derived in
+//! Python.
+class BHEInflowPythonBoundaryConditionPythonSideInterface
+{
+public:
+    /*!
+     * Initialize network dataframe
+     * return a tuple (time, BHE inflow temperature, BHE outflow
+     * temperature, BHE outflow bc node id, BHE flowrate)
+     * set at that position and the parameters of the BHE network.
+     */
+    virtual std::tuple<bool,
+                       double /*time*/,
+                       std::vector<double> /*Tin_val*/,
+                       std::vector<double> /*Tout_val*/,
+                       std::vector<int> /*bc_out_ids*/,
+                       std::vector<double> /*BHE_flowrate*/>
+    initializeDataContainer() const
+    {
+        _overridden_essential = false;
+        return std::tuple<bool, double, std::vector<double>, std::vector<double>,
+                          std::vector<int>,
+                          std::vector<double>>{false,
+                                   std::numeric_limits<double>::quiet_NaN(),
+                                   {}, {}, {}, {}};
+    }
+
+    /*!
+     * transfer BHE network dataframe to TESPy and get Tin from TESPy
+     *
+     * \return a tuple (time, BHE Tin and Tout value from TESPy)
+     * indicating if a natural BC shall be set at that position and the new
+     * inflow temperature of all BHEs
+     */
+    virtual std::tuple<bool, bool, std::vector<double>> tespyThermalSolver(
```

Comment:
The tuple types do not match with the documentation.
Could be improved also by creating a small `struct` for the return type.
At least fix the comment.

Reply:
In [658381](https://github.com/ufz/ogs/pull/2702/commits/6583812c9c2c34311b9328de578a3a8b64acc668), the comments have been fixed.